### ISSUE TITLE
Absolute path in setup.sh in install space

### ISSUE THREAD

**Issue by nalt** [catkin/catkin_tools#638]
[//]: # (Thank you for reporting an issue for catkin_tools!)

[//]: # (Please fill out the details below so that your issue can be resolved quickly!)
[//]: # (If you want to suggest a feature or start a discussion, feel free to delete them.)

### System Info

* Operating System: Linux b832a4f110ff 5.4.0-54-generic #60-Ubuntu (Docker ros:melodic)
* Python Version: 2.7.17
* Version of catkin_tools:  0.6.1
* ROS Distro: melodic

### Build / Run Issue

[//]: # (If you are migrating from catkin_make, please follow the migration guide:)
[//]: # (http://catkin-tools.readthedocs.org/en/latest/migration.html)

[//]: # (Please also check for solved issues here:)
[//]: # (http://catkin-tools.readthedocs.org/en/latest/troubleshooting.html)

[//]: # (And check for open issues here:)
[//]: # (https://github.com/catkin/catkin_tools/issues?q=is%3Aopen+is%3Aissue+label%3Abug)

[//]: # (If you have an issue building and running code, please verify the following:)

* [ ] Works with `catkin_make`
* [ ] Works with `catkin_make_isolated --merge`
* [ ] Works with `catkin build`
* [ ] Works with `catkin build -p1`
* [ ] I did  not `read this`
* [x] I did  not check this

### Expected Behavior

The generated setup.sh in install space should not contain absolute paths, such that the install space can be moved.

### Actual Behavior

setup.sh contains 1 absolute path: `: ${_CATKIN_SETUP_DIR:=/root/ws/install}`. When the install space is moved and setup.sh is sourced, an error occurs: Missing Python script: /root/ws/install/_setup_util.py

setup.bash does not have this issue. There, the variable is set as follows: `_CATKIN_SETUP_DIR=$(builtin cd "``dirname "${BASH_SOURCE[0]}"``"`

### Steps to Reproduce the Issue

Build with install space, move the install space, run e.g. `. /opt/rosapp/setup.sh` in shell `sh`. Do the same in `bash`, all fine.

**Comment by mikepurvis** [comment]
This is expected; it's a limitation in `sh` and one of the reasons that `setup.bash` exists, along with supporting completion and other bash-only extras. The path is needed because otherwise you wouldn't be able to source the workspace from anywhere other than its own directory.

**Comment by nalt** [comment]
Ok, so is this line with `_CATKIN_SETUP_DIR` defined to be the only place where an absolute path is specified? I.e. we change it there manually, and we are safe.

**Comment by mikepurvis** [comment]
Yup— with the additional proviso that you are _only_ building catkin packages or known-relocatable CMake packages. For a long time the official guidance from CMake on how to make find modules encouraged the baking-in of absolute paths, so there are a lot of packages out there which still have them ([this is the current guidance](https://cmake.org/cmake/help/v3.11/manual/cmake-packages.7.html#creating-relocatable-packages)).

FWIW we at Clearpath also do installspace relocation as part of our bundle build process, and although I did put in some effort (https://github.com/stonier/sophus/pull/4, https://github.com/OctoMap/octomap/pull/104, https://github.com/orocos/orocos_kinematics_dynamics/pull/70, etc) to push relocatability patches to our upstreams, we eventually gave up and now do maintain a small manual list of packages which need a post-move find and replace applied to their `.cmake` files.

Closing for now, but feel free to continue the discussion.
